source/detail/parser/x86_64.cpp

```diff
@@ -358,24 +357,9 @@ namespace smeagle::x86_64 {
         parameter p;
         p.name = paramName;
         p.type = paramType->getName();
-        std::string location;
-
-        // If we have a RegisterClass::MEMORY we will be using a framebase
-        if (regClasses.first == RegisterClass::MEMORY) {
-          location = "framebase+" + std::to_string(framebase);
-
-          // Update the framebase for the next parameter based on the type
-          framebase = updateFramebaseFromType(paramType, framebase);
-
-          // Otherwise we generate a string from register classes
-        } else {
-          location = getRegistersString(regClasses, order);
-        }
-
+        p.location = allocator.getRegistersString(regClasses, paramType);
```

Comment:
Oh, yeah, that's nice. No more logic at the call site!

Reply:
I thought so too :)